memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java

```diff
@@ -325,7 +327,14 @@ public void deleteConversation(String conversationId, ActionListener<Boolean> li
                 }, listener::onFailure);
 
             } else {
-                listener.onResponse(false);
+                log.error("No access to delete the memory for " + conversationId);
```

Comment:
From reading the code it seems like when we don't have have access to that conversation we mark this as "Resource not found exception". 

Is that the correct understanding? If yes, that doesn't seem quite right.

Reply:
The Olly chatbot pen tester thinks that UNAUTHORIZED(401) would suggests the user that the memory id at least exists, which is considered as leaking information. So returning "Not Found"(404) is a better option because it only indicates that you cannot find it.

Reply:
I see, makes sense. Can we add a comment in the code then? So that later we don't find it confusing again?